pandaserver/dataservice/closer.py

```diff
@@ -0,0 +1,465 @@
+                                "cleanup",
+                                "tobeclosed",
+                            ]:
+                                tmp_log.debug(
+                                    f"skip {unmerged_dataset_name} due to status={unmerged_dataset.status}")
+                            else:
+                                # set status
+                                unmerged_dataset.status = final_status
+                                # append to avoid repetition
+                                self.top_user_dataset_list.append(unmerged_dataset_name)
+                                # update DB
+                                ret_top_t = self.task_buffer.updateDatasets(
+                                    [unmerged_dataset],
+                                    withLock=True,
+                                    withCriteria="status<>:crStatus",
+                                    criteriaMap={":crStatus": unmerged_dataset.status},
+                                )
+                                if len(ret_top_t) > 0 and ret_top_t[0] == 1:
+                                    tmp_log.debug(
+                                        f"set {unmerged_dataset.status} to parent dataset : {unmerged_dataset_name}")
+                                else:
+                                    tmp_log.debug(
+                                        f"failed to update parent dataset : {unmerged_dataset_name}")
+        # start Activator
+        if re.search("_sub\d+$", dataset.name) is None:
+            if self.job.prodSourceLabel == "panda" and self.job.processingType in ["merge", "unmerge"]:
+                # don't trigger Activator for merge jobs
+                pass
+            else:
+                if self.job.jobStatus == "finished":
+                    activator_thread = Activator(self.task_buffer, dataset)
+                    activator_thread.start()
+                    activator_thread.join()
+        else:
+            # unset flag since another thread already updated
+            # flag_complete = False
+            pass
+
+    # main
+    def run(self):
+        """
+        Main method to run the Closer class. It processes each destination dispatch block,
+        updates the dataset status, finalizes pending jobs if necessary, and starts the notifier.
+        """
+        try:
+            tmp_log = LogWrapper(_logger,
+                                 f"run-{datetime.datetime.now(datetime.timezone.utc).replace(tzinfo=None).isoformat('/')}-{self.panda_id}")
+            tmp_log.debug(f"Start {self.job.jobStatus}")
+            flag_complete = True
+            final_status_dataset = []
+
+            for destination_dispatch_block in self.destination_dispatch_blocks:
+                dataset_list = []
+                tmp_log.debug(f"start {destination_dispatch_block}")
+
+                # ignore tid datasets
+                if re.search("_tid[\d_]+$", destination_dispatch_block):
```

Comment:
is_tid_dataset function? What's a tid dataset? We need to add this to the documentation together with the special agent configuration.

Reply:
task output dataset